### DISCUSSION TITLE
Tailwind support

### DISCUSSION THREAD

**Discussion by thesobercoder** [mint-lang/mint#348] [Q&A]
I'm trying out mint and so far I'm loving it. The live code rebuilds are ridiculously fast. Would be great to see official Tailwind CSS support.

**Comment by gdotdesign** [comment]
Hi, I'm glad you like Mint :slightly_smiling_face: 

I'm not sure what you mean by official support, what you can do now is to include the compiled Tailwind CSS in either [the head file](https://www.mint-lang.com/guide/recipes/using-third-party-css) or in the `external.stylesheets` of the `mint.json`.

**Comment by iambudi** [comment]
I guess official support means ability to tree-shake unused tailwind styles (purge) and optimize build size?

**Comment by nailuj29** [comment]
you can see https://www.mint-lang.com/guide/recipes/using-third-party-css for info on how to set it up

**Comment by bstro** [comment]
It is unclear to me, but it seems that the guide does not address tree-shaking unused tailwind styles. First-class Tailwind support would be a very attractive feature to me.

**Comment by gdotdesign** [comment]
> I guess official support means ability to tree-shake unused tailwind styles (purge) and optimize build size?

That's not possible at the moment because Mint does not interface with the NPM ecosystem, until it does the only option is to use a pre-build version of it.

Also it's not possible to create tailwind style utility CSS because there are no global styles.

**Reply by techgems** [reply]
If you use postcss directly you won't need to interact with NPM at all. It'd also be a good way to add the ability for global styles in a way that doesn't force new design choices on the language.

It'd be having an opinionated use (yet optional) on postcss that on build process would generate something to feed to the third party css.

I'm learning Mint and having stronger TailwindCSS support, would be an amazing plus for me to finally ditch React.

**Comment by aislanmaia** [comment]
I'm looking for that language now and cannot see how to use Tailwind (JIT version) with [mint-ui](https://github.com/mint-lang/mint-ui), for example.

**Comment by DavidBernal** [comment]
Hi, I created a repo to show how integrate mint + tailwind with tree-shaking:

https://github.com/DavidBernal/mint-tailwind-boilerplate

Probably is not the best solution, but is easy and works correctly.
